### PR TITLE
fix: check updatability before updating DO InRef binding

### PR DESCRIPTION
**Pull request by AliouDIAITE**

### CODE REVIEW

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java [hunk 1]
```diff
@@ -153,6 +159,21 @@ private String createInRefValTestString(TExtRef extRef) {
         return new ObjectReference(extRef, ExtrefTarget.SRC_CB).getReference();
     }
 
+    private ResumedDataTemplate createDaiFilter(String daName) {
```
Comment (samirromdhani) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java:162]: This instance of ResumedDataTemplate can be created in ResumedDataTemplate class
```java
    public ResumedDataTemplate (AbstractLNAdapter lnAdapter, String doName, String daName) {
        this.lnClass = lnAdapter.getLNClass();
        this.lnInst = lnAdapter.getLNInst();
        this.prefix = lnAdapter.getPrefix();
        this.lnType = lnAdapter.getLnType();
        this.doName = new DoTypeName(doName);
        this.daName = new DaTypeName(daName);
    }
```

**Review by samirromdhani — APPROVED** [review]

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java [hunk 2]
```diff
@@ -141,6 +136,17 @@ public Optional<SclReportItem> updateDaiFromExtRef(List<TExtRef> tExtRefs) {
         return Optional.empty();
     }
 
+    private void updateDAI(String daName, String value) {
+        ResumedDataTemplate daiFilterSrcRef = createDaiFilter(daName);
+        List<ResumedDataTemplate> list1 = getParentAdapter().getDAI(daiFilterSrcRef, true); // Should find oonly one DAI
```
Comment (massifben) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java:141]: I suggest renaming `list1` to `foundDais`

File: sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapterTest.java [hunk 3]
```diff
@@ -569,4 +565,70 @@ void updateDaiFromExtRef_should_throw_exception_when_ExtRef_desc_dont_end_with__
         assertThatThrownBy(() -> doiAdapter.updateDaiFromExtRef(extRefList))
                 .isInstanceOf(NumberFormatException.class);
     }
+
+    private DOIAdapter createDOIAdapteInScl() {
```
Comment (massifben) [inline sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapterTest.java:569]: typo : `createDOIAdapterInScl`

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (AliouDIAITE) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java:141]: we are looking if DA is present and updatable either in IED section or in DataTypeTemplate section. getDAI methode return a list depending on what e specify in the filter. In our case e ask for a specific DA either it is or it is not know. Thats wh getDAI response should be list with 1 element or empty list.
Ok for renaming list1

[the remark this answers is not in the document]
Comment (massifben) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/DOIAdapter.java]: I understand. Can you update the comment `// Should find oonly one DA`  by something like `// Check if if DA is present and updatable` ?

**Pull request merged**